File: bots_and_utilities/bot.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

// Функция для добавления клиента
func addClient(name, formula string) {
    data := map[string]string{"name": name, "formula": formula}
    jsonData, err := json.Marshal(data)
    if err != nil {
        fmt.Println("Error marshalling data:", err)
        return
    }

    response, err := http.Post("http://localhost:5000/add_client", "application/json", bytes.NewBuffer(jsonData))
    if err != nil {
        fmt.Println("Error sending request:", err)
        return
    }
    defer response.Body.Close()
    body, err := ioutil.ReadAll(response.Body)
    if err != nil {
        fmt.Println("Error reading response:", err)
        return
    }
    fmt.Println("Add Client Response:", string(body))
}

// Функция для расчета выплаты
func calculatePayment(clientName string, amount float64) {
    data := map[string]interface{}{"client_name": clientName, "amount": amount}
    jsonData, err := json.Marshal(data)
    if err != nil {
        fmt.Println("Error marshalling data:", err)
        return
    }

    response, err := http.Post("http://localhost:5000/calculate_payment", "application/json", bytes.NewBuffer(jsonData))
    if err != nil {
        fmt.Println("Error sending request:", err)
        return
    }
    defer response.Body.Close()
    body, err := ioutil.ReadAll(response.Body)
    if err != nil {
        fmt.Println("Error reading response:", err)
        return
    }
    fmt.Println("Calculate Payment Response:", string(body))
}

func main() {
    // Добавляем клиента с формулой (salary * 12) * 0.18 - 6%
    addClient("ExampleClient1", "(amount*12)*0.18-6")

    // Рассчитываем выплату для клиента "ExampleClient" с зарплатой 100
    calculatePayment("ExampleClient1", 100)
}

File: bots_and_utilities/bot.py
import requests

BASE_URL = "http://localhost:5000"

def add_client():
    name = input("Enter client name: ")
    formula = input("Enter formula: ")
    data = {"name": name, "formula": formula}
    response = requests.post(f"{BASE_URL}/add_client", json=data)
    print(f"Status Code: {response.status_code}")
    print(response.json())

def delete_client():
    name = input("Enter client name to delete: ")
    response = requests.delete(f"{BASE_URL}/delete_client/{name}")
    print(f"Status Code: {response.status_code}")
    print(response.json())

def add_admin():
    id = input("Enter admin id: ")
    data = {"id": id}
    response = requests.post(f"{BASE_URL}/add_admin", json=data)
    print(f"Status Code: {response.status_code}")
    print(response.json())

def remove_admin():
    id = input("Enter admin id to remove: ")
    response = requests.delete(f"{BASE_URL}/remove_admin/{id}")
    print(f"Status Code: {response.status_code}")
    print(response.json())

def get_all_admins():
    response = requests.get(f"{BASE_URL}/get_all_admins")
    print(f"Status Code: {response.status_code}")
    print(response.json())

def update_client_formula():
    name = input("Enter client name: ")
    new_formula = input("Enter new formula: ")
    data = {"name": name, "new_formula": new_formula}
    response = requests.put(f"{BASE_URL}/update_client_formula", json=data)
    print(f"Status Code: {response.status_code}")
    print(response.json())

def calculate_payment():
    client_name = input("Enter client name: ")
    payment = input("Enter payment: ")
    data = {"client_name": client_name, "payment": payment}
    response = requests.post(f"{BASE_URL}/calculate_payment", json=data)
    print(f"Status Code: {response.status_code}")
    print(response.json())

def get_all_clients():
    response = requests.get(f"{BASE_URL}/get_all_clients")
    print(f"Status Code: {response.status_code}")
    print(response.json())

def main():
    while True:
        print("\nMenu:")
        print("1. Add Client")
        print("2. Delete Client")
        print("3. Update Client Formula")
        print("4. Calculate Payment")
        print("5. Get All Clients")
        print("6. Add admin")
        print("7. Delete admin")
        print("8. Get All Clients")
        print("9. Exit")
        
        choice = input("Enter your choice: ")
        
        if choice == '1':
            add_client()
        elif choice == '2':
            delete_client()
        elif choice == '3':
            update_client_formula()
        elif choice == '4':
            calculate_payment()
        elif choice == '5':
            get_all_clients()
        elif choice == '6':
            add_admin()
        elif choice == '7':
            remove_admin()
        elif choice == '8':
            get_all_admins()
        elif choice == '9':
            print("Exiting...")
            break
        else:
            print("Invalid choice, please try again.")

if __name__ == "__main__":
    main()
